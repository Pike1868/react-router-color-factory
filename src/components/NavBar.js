import React from "react";
import { Link } from "react-router-dom";

/**
 * AddColorForm Component:
 *
 * Provides navigation for app, displays welcome message and a link to the form for adding a new color.
 *
 */

const NavBar = () => {
  return (
    <nav
      style={{
        padding: "50px",
        backgroundColor: "darkgray",
        textAlign: "center",
      }}
    >
      <h3>Welcome to Color Factory</h3>
      <Link to="/colors/new" style={{ fontSize: "1.5rem" }}>
        Add a color
      </Link>
    </nav>
  );
};

export default NavBar;
