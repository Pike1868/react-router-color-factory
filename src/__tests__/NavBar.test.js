import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../components/NavBar";

//Smoke Test
test("NavBar component renders without crashing", () => {
  render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );
});

//Snapshot Test
test("NavBar component matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();

  const navHeading = screen.getByText("Welcome to Color Factory");

  expect(navHeading).toBeInTheDocument();
});
