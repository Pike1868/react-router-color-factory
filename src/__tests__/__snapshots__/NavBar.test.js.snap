// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    style="padding: 50px; background-color: darkgray; text-align: center;"
  >
    <h3>
      Welcome to Color Factory
    </h3>
    <a
      href="/colors/new"
      style="font-size: 1.5rem;"
    >
      Add a color
    </a>
  </nav>
</DocumentFragment>
`;
